src/libraries/Microsoft.PowerFx.Core/Public/Types/UserProvidedTypeDefinitions/FormulaTypeToSchemaHelper.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core
+{
+    internal static class FormulaTypeToSchemaHelper
+    {
+        public static FormulaTypeSchema ToSchema(this FormulaType type, DefinedTypeSymbolTable definedTypeSymbols)
+        {
+            return ToSchema(type, definedTypeSymbols, maxDepth: 5);
```

Comment:
>5) [](http://example.com/codeflow?start=64&length=2)

We probably need to document this limitation.

Reply:
Yes. At the moment, FormulaType.ToSchema is only going to be used for the FormulaBar response, so it's safe to restrict like this. We do need some restriction, as types can be recursive in a way FormulaTypeSchema may not be able to be.